starfish/test/test_multiprocessing.py

```diff
@@ -81,6 +84,64 @@ def _decode_wrapped_array_to_numpy_array(wrapped_array):
     return np.frombuffer(wrapped_array.array.get_obj(), dtype=np.uint8)
 
 
+def test_xr_deepcopy(nitems: int=10) -> None:
+    """
+    Instantiate an :py:class:`xarray.DataArray` and run
+    :py:method:`starfish.imagestack.mp_dataarray.xr_deepcopy` on it.  The copy is passed to worker
+    processes.  Worker processes reconstitute a numpy array from the buffer and attempts to writes
+    to the numpy array.  Writes in the worker process should be visible in the parent process in the
+    copy but not the original.
+    """
+    shape = (nitems, )
+    source = np.zeros(shape, dtype=np.uint8)
+    dataarray = xr.DataArray(source)
+    copy, buffer = _mp_dataarray.xr_deepcopy(dataarray)
+    _start_process_to_test_shmem(buffer, _decode_array_to_numpy_array, nitems)
+    for ix in range(nitems):
+        assert dataarray[ix] == 0
+        assert copy[ix] == ix
+
+
+def test_imagestack_deepcopy(nitems: int=10) -> None:
+    """
+    Instantiate an :py:class:`ImageStack` and deepcopy it.  Worker processes reconstitute a numpy
+    array from the buffer and attempts to writes to the numpy array.  Writes in the worker process
+    should be visible in the parent process.
+    """
+    shape = (nitems, 3, 4, 5, 6)
+    dtype = np.float32
+    source = np.zeros(shape, dtype=np.float32)
+    imagestack = ImageStack.from_numpy_array(source)
+    imagestack_copy = copy.deepcopy(imagestack)
+    _start_process_to_test_shmem(
+        imagestack_copy._data._backing_mp_array,
+        partial(_decode_imagestack_array_to_numpy_array, shape, dtype),
+        nitems)
+    for ix in range(nitems):
+        assert (imagestack.xarray[ix] == 0).all()
+        assert np.allclose(imagestack_copy.xarray[ix], ix)
+
+
+def _decode_imagestack_array_to_numpy_array(
+        shape: Tuple[int, ...], dtype, buffer) -> np.ndarray:
+    unshaped_np_array = np.frombuffer(buffer.get_obj(), dtype=dtype)
+    return unshaped_np_array.reshape(shape)
+
+
+def test_imagestack_xarray_deepcopy(nitems: int=10) -> None:
```

Comment:
Is it worth explicitly demonstrating that improperly copying does _not_ work with multiprocessing?

Reply:
If you don't use the proper copy, you don't have a `multiprocessing.Array` to pass to `multiprocessing.Pool`.  In that case, you're just passing a numpy array to `multiprocessing.Pool`, which is `test_numpy_array` (first case of `starfish/test/test_multiprocessing.py`)